domain/administrator.js

```diff
@@ -8,31 +8,95 @@ class Administrator {
 
   async viewDoctors() {
```

Comment:
To simplify the repository, we should make the differentiation for the database query in the domain model. in place of fetchPendingDoctors, we could have fetchPendingUser(<userId>, <userType>); this way we would have less code to test and the readability will also increase.

Reply:
What do you think?

Reply:
I just simplified all the routes for the admin and increased code reuse. Good catch!